src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -196,7 +198,26 @@ private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModul
             diagnostics.AddRange(entryPointAndDiagnostics.Diagnostics);
 
             var entryPoint = entryPointAndDiagnostics.MethodSymbol;
-            var synthesizedEntryPoint = entryPoint as SynthesizedEntryPointSymbol;
+
+            if ((object)entryPoint == null)
+            {
+                Debug.Assert(entryPointAndDiagnostics.Diagnostics.HasAnyErrors() || !compilation.Options.Errors.IsDefaultOrEmpty);
+                return null;
+            }
+
+            // True if a Main definition has been created and needs to be added to the module being built
```

Comment:
Is this comment associated with this declaration or with the `if` statement at line 200? If it's this declaration, perhaps "Non-null if ..." rather than "True if ..."